src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1040,6 +1040,12 @@ ExecuteTaskListIntoTupleDest(RowModifyLevel modLevel, List *taskList,
 uint64
 ExecuteTaskListExtended(ExecutionParams *executionParams)
 {
+	/* if there are no tasks to execute, we can return early */
```

Comment:
Why do we need that? Is that needed by this PR?

Reply:
This change is made in a commit 6c0e3cbcf91af00185cd0b63c6a988de6d7ca218, which I opened a separate PR for (https://github.com/citusdata/citus/pull/6900). I included that commit in that PR too to make the tests passing here as well.